File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> s = new Stack<>();
        int[] array = new int[temperatures.length];
        for(int i = 0; i< temperatures.length; i++){
            if(s.empty()){
                int[] a = new int[2];
                a[0] = temperatures[i];
                a[1] = i;
                s.push(a);
                continue;
            }
            if(temperatures[i] > s.peek()[0]){
                int[] elem = s.pop();
                array[elem[1]] = i - elem[1];
                i--;
            }else{
                int[] a = new int[2];
                a[0] = temperatures[i];
                a[1] = i;
                s.push(a);
            }
        }
        return array;
    }
}

File: Data Structures & Algorithms/implement-queue-using-stacks/submission-0.cpp
class MyQueue {
public:
        stack<int> a;
        stack<int> b;
    MyQueue() {
    }
    
    void push(int x) {
        a.push(x);
    }
    
    int pop() {
        while(!a.empty()){
            b.push(a.top());
            a.pop();
        }
        int i = b.top();
        b.pop();
        while(!b.empty()){
            a.push(b.top());
            b.pop();
        }
        return i;
    }
    
    int peek() {
       while(!a.empty()){
            b.push(a.top());
            a.pop();
        }
        int i = b.top();
        while(!b.empty()){
            a.push(b.top());
            b.pop();
        }
        return i;

    }
    
    bool empty() {
        return a.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */